### PR TITLE
fix: setqflist("all") should respect change_base

### PR DESCRIPTION
**Pull request by xzbdmw**
If change_base is called, instead of run git status, run git diff against the changed base_commit.
Fix: #692

### CODE REVIEW

File: lua/gitsigns/git/repo.lua [hunk 1]
```diff
@@ -44,18 +45,27 @@ function M:command(args, spec)
   return git_command(args1, spec)
 end
 
---- @return string[]
 function M:files_changed()
```
Comment (lewis6991) [inline lua/gitsigns/git/repo.lua:48]: Don't pull the config into this file. Instead add the sha as an argument to this function.

Comment (xzbdmw) [reply lua/gitsigns/git/repo.lua]: Done

File: lua/gitsigns/git/repo.lua [hunk 2]
```diff
@@ -44,18 +44,29 @@ function M:command(args, spec)
...
 end
 
+--- @param base string?
 --- @return string[]
-function M:files_changed()
+function M:files_changed(base)
   --- @type string[]
-  local results = self:command({ 'status', '--porcelain', '--ignore-submodules' })
+  local results
+  if base then
```
[the remark this answers is not in the document]
Comment (xzbdmw) [reply lua/gitsigns/git/repo.lua:52]: When will base be :0?

[the remark this answers is not in the document]
Comment (lewis6991) [reply lua/gitsigns/git/repo.lua:52]: When someone sets it.

[the remark this answers is not in the document]
Comment (xzbdmw) [reply lua/gitsigns/git/repo.lua:52]: Hmm, it seems base will never be :0
https://github.com/lewis6991/gitsigns.nvim/blob/f074844b60f9e151970fbcdbeb8a2cd52b6ef25a/lua/gitsigns/util.lua#L237

[the remark this answers is not in the document]
Comment (lewis6991) [reply lua/gitsigns/git/repo.lua:52]: Doesn't matter, it might not in the future.

[the remark this answers is not in the document]
Comment (xzbdmw) [reply lua/gitsigns/git/repo.lua]: I add an additional check in both places, is that enough?

**Comment by lewis6991** [comment]
Commits needs squashing.

**Comment by lewis6991** [comment]
Thanks

**Pull request merged**